### DISCUSSION TITLE
How to set the Apple deployment target when using manifest mode?

### DISCUSSION THREAD

**Discussion by MarkCallow** [microsoft/vcpkg#39966] [Q&A]
In my project I am getting more than 330 linker warnings like this

```
Object file (<project dir>/build/ios/vcpkg_installed/arm64-ios/lib/libassimp.a[4](Compression.cpp.o)) was built for newer 'iOS' version (17.5) than being linked (12.0)
```

In my project's CMakeLists.txt I am setting the `CMAKE_OSX_DEPLOYMENT_TARGET` to "12.0". These warnings happened immediately I got vcpkg manifest mode set up for the project. I thought the problem was likely that I was setting `CMAKE_OSX_DEPLOYMENT_TARGET` after the `project` command so I moved the `project` command to after the setting. It made no difference. After that I set `VCPKG_OSX_DEPLOYMENT_TARGET` to the value of `CMAKE_OSX_DEPLOYMENT_TARGET` still before `project`. That too made no difference.

How can I get vcpkg to compile dependencies for the deployment target chosen by my project?

My web searches so far have proved fruitless.

**Comment by dg0yt** [comment]
`VCPKG_OSX_DEPLOYMENT_TARGET` is a triplet variable, i.e. you might need to customize your triplet (file). This would not only affect the actual port builds but also the ABI hashes, i.e. determining what needs to be rebuilt and what can be reused.

**Reply by MarkCallow** [reply]
I do not have a triplet file. How do I add one? My project can be built for multiple architectures and platforms from the same Git workarea. Does that mean I need multiple triplet files and have CMake configuration pick the right one? If so, how do I do that?

This sounds like a lot of work, especially as vcpkg is currently automatically selecting the right triplets for platform and architecture without me doing anything. Maybe I should file an enhancement request asking for vcpkg to pay attention to `CMAKE_OSX_DEPLOYMENT_TARGET` instead of always selecting for the current OS version.

**Reply by dg0yt** [reply]
You use a triplet file implicitly, probably `triplets/community/arm64-ios.cmake`. 
You might choose another triplet name, or you can use a modified copy with `--overlay-triplets=...`.

Given that even `CMAKE_CXX_COMPILER`, `CMAKE_SYSTEM_VERSION` or `VCPKG_HOST_TRIPLET` etc. are not passed from the CMake project to the ports installation, `CMAKE_OSX_DEPLOYMENT_TARGET` isn't treated worse...

**Reply by MarkCallow** [reply]
Thank you. I have got it working (for iOS for now). I find this very user unfriendly. I have to copy a minimum of 4 triplet files for iOS and OSX, edit each of them to add `VCPKG_OSX_DEPLOYMENT_TARGET` to a target matching the `CMAKE_OSX_DEPLOYMENT_TARGET` in my CMakeLists.txt. If I need to change the deployment target in future I have to change it in 5 files all the while potentially missing updates to the source triplet files. I will open an issue.
